### ISSUE TITLE
Generated exe does not run when using generated binary package

### ISSUE THREAD

**Issue by AESilky** [Nuitka/Nuitka#2845]
> Issue created to continue to address problem originally reported in #2841

> Moved here from Discord discussion at the request of @kayhayen
> AES started discussion at 05/04/2024 8:26PM
> Responded to by KeyHayen and others through 05/07/2024 10:42AM

Generated executables, when run from a prompt, are silent for about 3 seconds and then return to the prompt, when the executable is built to rely on a binary package also created using Nuitka.

**What does work:**
If the executable is created using the .py version of the package (such that the package/modules get slurped up into the executable) it runs fine. If the .py version of the application is run using the binary package it runs fine.

**Comment by AESilky** [comment]
Using one of the smaller utility applications, with the package and application generated with `--debug --python-flag=-v`.
The following is found in the *err.txt* output:
```
import 'platform' # <class 'nuitka_module_loader'>
import socket # considering responsibility (find_spec)
import socket # denied responsibility
import socket # considering responsibility (find_spec)
import socket # denied responsibility
Traceback (most recent call last):
  File "C:\Users\aesil\proj\nuitka\PyKOB\bin\Sample.dist\Sample.py", line 38, in <module>
  File "<frozen importlib._bootstrap>", line 1232, in _handle_fromlist
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1147, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 690, in _load_unlocked
  File "C:\Users\aesil\proj\nuitka\PyKOB\bin\Sample.dist\pykob\config.py", line 49, in <module pykob.config>
  File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1140, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'socket'
# destroy pykob.config
```

**Comment by AESilky** [comment]
In my **es-nuitka** branch, 
https://github.com/MorseKOB/PyKOB/tree/es-nuitka

I added:
```
nuitka_sample_bld.cmd
nuitka_mkob_bld.cmd
nuitka_mrt_bld.cmd
```
In the root. The whole repo is public.

These build the 'Sample', 'MKOB', and 'MRT' applications, each of which use the 'pykob' package that is also generated through Nuitka (it is built by those .cmd files as well).

Yes, the 'factory' version of Nuitka is being used. Yes, the '.pyi' file is copied into the src.py directory before the builds of the applications.

**Sample** is a command-line utility application that people use to see that things are working. It is small and uses the _pykob_ package/modules.
**MKOB** is a tkinter GUI application. It is the primary application in the suite.
**MRT** is a command-line application that does a lot of what MKOB does, but without using a GUI.

These can be run using Python to compare:
```
cd src.py
python Sample.py

python MRT.py

python MKOB.py
```

I will want all of these, and the rest of the Python files in the src.py directory that start with a capital letter, to build into executables to be distributed as the PyKOB suite.

**Comment by kayhayen** [comment]
Directories named `src.py` could actually be an interesting challenge to Nuitka as an aside. :-)

**Comment by kayhayen** [comment]
The `Sample` one just worked for me with these options: `--run --stand --include-package-data=pykob Sample.py` and I am now looking at our build script.

As an aside, use `python setup.py develop` to install Nuitka from git into your Python and call it as `python -m nuitka` as that is going to make it more portable, no need to hard code paths.

When you use `--nofollow-import-to=pykob` it will be as if `pykob` is not there, which you seem to make up for by copying the Python files to the distribution, something which is not supported. Dependencies are not analysed, and the extension module doesn't get used. I forget which takes precedence, the pykob package directory of the extension module all the time, when creating it Nuitka warns if it's a problem.

I also noticed, you use a bunch of options, like --enable-console and --forcing outputs when creating the extension module, that will absolutely have no effect.

Basically you are not doing at all what you were saying, you are copying Python code that Nuitka has not dependency analysed, and if that is what you want, you can do that of course, but you get to manually maintain the dependencies. Adding them to the extension module was also never a good idea, since stdlib is not supposed to live in extension modules. And you would have had to add them to the exe creates.

**Comment by AESilky** [comment]
Changing the source tree such that the `pykob` module source is in a separate tree and then removing the `--nofollow-import-to=pykob`, then copying the `pykob.pyd` and `pykob.pyi` files into the _src.py_ directory (where the rest of the applications' .py files are) results in executables that work correctly.

**Comment by AESilky** [comment]
When will the changes in _factory_ move into the main branch so I can use it?

**Comment by kayhayen** [comment]
The release process regarding factory is explained here https://nuitka.net/doc/factory.html
